haystack/nodes/retriever/dense.py

```diff
@@ -493,12 +486,19 @@ def _get_predictions(self, dicts):
             leave=False,
             disable=disable_tqdm,
         ) as progress_bar:
-            for batch in data_loader:
-                batch = {key: batch[key].to(self.devices[0]) for key in batch}
+            for raw_batch in data_loader:
+                batch = {key: raw_batch[key].to(self.devices[0]) for key in raw_batch}
 
                 # get logits
                 with torch.no_grad():
-                    query_embeddings, passage_embeddings = self.model.forward(**batch)[0]
+                    query_embeddings, passage_embeddings = self.model.forward(
```

Comment:
Not sure what is better here: silent failures in case there are no parameters `query_input_ids` etc or fail-fast exception if we don't use get. Thoughts?

Reply:
I think we need to allow them to be `None` in here. `self.model` is a bit of a misleading name: that is not a `LanguageModel`, but a `BiAdaptiveModel`, whose `forward` method will check which vectors are given and behave differently according to what it receives.

I personally don't like this architecture at all (one never knows what will come out of the same method...) but I can't change everything in one go :sweat_smile: I guess this has to stay unless you have something in mind.

Reply:
Deal, understood, make sense. Thanks for the details @ZanSara